src/main/kotlin/com/zepben/evolve/services/network/EquivalentNetworkUtils.kt

```diff
@@ -0,0 +1,465 @@
+     * These edge [ConnectivityNode] are identified by having a single [Terminal] that has a [ConductingEquipment] that belongs to both neighbouring [EquipmentContainer]s.
+     *
+     * This overload of the function takes two [EquipmentContainer] instances to identify the edge [ConnectivityNode]s.
+     *
+     * @param container The [EquipmentContainer] that has its equipment in the [NetworkService] (it doesn't hold the edge [ConnectivityNode]).
+     * @param otherContainer The edge [EquipmentContainer] that doesn't have its equipment in the [NetworkService] (it holds the edge [ConnectivityNode]).
+     * @param network The [NetworkService] that we need to add the [EquivalentBranch] and [ConductingEquipment] to.
+     * @param createEquivalentBranches factory function to create the [EquivalentBranch]es that will be attached to the edge [ConnectivityNode] to represent the 'equivalent network'.
+     * @param createEquivalentEquipment factory function to create the [ConductingEquipment]s that will be attached to the [EquivalentBranch]es of the 'equivalent network'.
+     * @param maxNumber The maximum number of 'equivalent networks' you want to add.
+     *
+     * @return a set of [EquivalentNetworkConnection]s that hold information and references to the edge [ConnectivityNode], edge [ConductingEquipment], and added [EquivalentBranch]es and their [ConductingEquipment].
+     */
+    fun addToEdgeBetweenContainers(
+        container: EquipmentContainer,
+        otherContainer: EquipmentContainer,
+        network: NetworkService,
+        createEquivalentBranches: (EquivalentBranchDetails) -> Sequence<Pair<PhaseCode, EquivalentBranch>>,
+        createEquivalentEquipment: (EquivalentEquipmentDetails) -> Sequence<Pair<PhaseCode, ConductingEquipment>>,
+        maxNumber: Int? = null
+    ): Set<EquivalentNetworkConnection> {
+        val edgeCnn = getEdgeNodes(network, setOf(container::class, otherContainer::class), container, otherContainer)
+        val edgeCnToProcess = (maxNumber?.let { edgeCnn.take(it) } ?: edgeCnn)
+        return edgeCnToProcess.map { cn -> addEquivalentNetwork(network, cn, createEquivalentBranches, createEquivalentEquipment) }.toSet()
+    }
+
+    /**
+     * Function to add 'equivalent networks' which are groups of [ConductingEquipment] connected through [EquivalentBranch]es that represent simplified networks.
+     * These 'equivalent networks' are attached to edge [ConnectivityNode]s, which are [ConnectivityNode]s right at the edge between [EquipmentContainer]s.
+     * These edge [ConnectivityNode] are identified by having a single [Terminal] that has a [ConductingEquipment] that belongs to both neighbouring [EquipmentContainer]s.
+     *
+     * This overload of the function takes an [EquipmentContainer] instance and an [EquipmentContainer] class reference to identify the edge [ConnectivityNode]s.
```

Comment:
Can you add more detail here to indicate what it will do with the class? i.e will it add equivalent networks at the edge of every network that matches the given class? Does maxNumber effect this?

Reply:
Done